use-timescale/migration/troubleshooting.md

```diff
@@ -0,0 +1,227 @@
+trigger which ensures that no database user can create or modify jobs owned by
+another database user. This trigger can provide an obstacle for migrations.
+
+If the `--no-owner` flag is used with pg_dump and pg_restore, all objects in the
+target database will be owned by the user that ran pg_restore, likely tsdbadmin.
+
+If all the background jobs in the source database were owned by a user of the
+same name as the user running pg_restore (again likely tsdbadmin), then loading
+the `_timescaledb_config.bgw_jobs` table should work.
+
+If the background jobs in the source were owned by the postgres user, they will 
+be automatically changed to be owned by the tsdbadmin. In this case, one just 
+needs to verify that the jobs do not make use of privileges that the tsdbadmin
+user does not possess.
+
+If background jobs are owned by one or more users other than the user employed 
+in restoring, then there could be issues. To work around this issue, do not dump
+this table with pg_dump. Provide either
+`--exclude-table-data='_timescaledb_config.bgw_job'` or 
+`--exclude-table='_timescaledb_config.bgw_job'` to pg_dump to skip this table.
+Then, use psql and the COPY command to dump and restore this table with modified
+values for the `owner` column.
+
+```bash
+# dump the _timescaledb_config.bgw_job table to a csv file replacing the owner 
+# values with tsdbadmin
+psql -d "$SOURCE" -X -v ON_ERROR_STOP=1 --echo-errors -f - <<'EOF'
+begin;
+select string_agg
+( case attname
+    when 'owner' then $$'tsdbadmin' as "owner"$$
+    else format('%I', attname)
+  end
+, ', '
+) as cols
+from pg_namespace n
+inner join pg_class c
+on (n.nspname = '_timescaledb_config'
+and n.oid = c.relnamespace
+and c.relname = 'bgw_job')
+inner join pg_attribute a
+on (c.oid = a.attrelid and a.attnum > 0)
+\gset
+copy
+(
+    select :cols 
+    from _timescaledb_config.bgw_job
+    where id >= 1000
+) to stdout with (format csv, header true)
+\g bgw_job.csv
+rollback;
+\q
+EOF
+
+# copy the csv file into the target's _timescaledb_config.bgw_job
+psql -X -d "$TARGET" -v ON_ERROR_STOP=1 --echo-errors \
+    -c "\copy _timescaledb_config.bgw_job from 'bgw_job.csv' with (format csv, header match)"
+```
+
+Once the table has been loaded and the restore completed, you may then use SQL
+to adjust the ownership of the jobs and/or the associated stored procedures and 
+functions as you wish.
+
+## Extension Availability
+
+There are a vast number of PostgreSQL extensions available in the wild. 
+Timescale supports many of the most popular extensions, but not all extensions.
+Before migrating, check that the extensions you are using are supported on
+Timescale. Use the query below in a Timescale instance to get an up-to-date list
+of supported extensions.
+
+```sql
+select x.*
+from pg_available_extensions x
+inner join
+(
+    select unnest(regexp_split_to_array(setting, ',')) as name
+    from pg_settings
+    where name = 'extwlist.extensions'
+) w on (x.name = w.name)
+```
```

Comment:
@jgpruitt as far as I know this is not correct. The official source of truth is this document: https://docs.timescale.com/use-timescale/latest/extensions/.

One example which _is_ in extwlist, but doesn't work, is `postgres_fdw`.

Reply:
😵 It's highly annoying that this information isn't available via SQL.